tests/python/test_python_frontend.py

```diff
@@ -4086,6 +4086,108 @@ def fusion_func(fd: FusionDefinition) -> None:
 
         nvf_out, _ = self.exec_nvfuser(fusion_func, inputs)
 
+    # Test that the range of generated uniform values spans the proper range
+    # https://github.com/NVIDIA/Fuser/issues/1653
+    def test_uniform_range(self):
+        dtypes = [DataType.Double, DataType.Float, DataType.Half]
+        if not is_pre_ampere():
+            dtypes.append(DataType.BFloat16)
+
+        def run_test(left: float, right: float, dtype: DataType):
+            samples_per_run = 2**29
+
+            def fusion_fn(fd: FusionDefinition):
+                # Generate enough values to reasonably expect to sample the ends of the range
+                shape = fd.define_vector([samples_per_run], dtype=DataType.Int)
+                S0 = fd.define_scalar(left, dtype=DataType.Double)
+                S1 = fd.define_scalar(right, dtype=DataType.Double)
+                output = fd.ops.uniform(S0, S1, shape=shape, dtype=dtype)
+                fd.add_output(output)
+
+            with FusionDefinition() as fd:
+                fusion_fn(fd)
+
+            output = fd.execute([])[0]
+
+            x = output.amax()
+            m = output.amin()
+            mu = output.type(torch.float64).mean()
+            # Repeat to improve chances of sampling extreme values
+            num_runs = 100
+            num_samples = num_runs * samples_per_run
+            for i in range(num_runs):
+                u = fd.execute([])[0]
+                x = torch.maximum(x, u.amax())
+                m = torch.minimum(m, u.amin())
+                mu = mu + (u.type(torch.float64).mean() - mu) / (i + 2)
+
+            # round-trip cast to find expected min
+            theomin = torch.tensor(left, dtype=output.dtype).item()
+            theomu = 0.5 * (right + left)
+            theomax = torch.nextafter(
+                torch.tensor(right, dtype=output.dtype),
+                torch.tensor(left, dtype=output.dtype),
+            )
+
+            if dtype != DataType.Double:
+                # Even with repeats, there's no hope of sampling extreme double values
```

Comment:
Yeah there are just many more values to sample because of the increased precision. Testing would be easier if we wrote these functions in the IR because then we could more easily test the extreme values.

Reply:
Of course, to get these functions into the IR we will need a half precision `eq` 😂